modules/git/blame.go

```diff
@@ -100,8 +101,9 @@ func CreateBlameReader(repoPath, commitID, file string) (*BlameReader, error) {
 }
 
 func createBlameReader(dir string, command ...string) (*BlameReader, error) {
-	// FIXME: graceful: This should have a timeout
-	ctx, cancel := context.WithCancel(DefaultContext)
+	// This timeout was arbitrarily chosen just so that there are not hundreds of `git blame`
+	// processes sticking around after some operations.
+	ctx, cancel := context.WithTimeout(DefaultContext, 5*time.Minute)
```

Comment:
Maybe use git command timeout setting?

Reply:
I'm not sure if that makes sense? Semantically they're rather different.

Reply:
I mean from these values: https://github.com/go-gitea/gitea/blob/master/custom/conf/app.example.ini#L910

Maybe new setting should be added for this